syntax = "proto3";

package flowpipe.stages.nats.jetstream.source.v1;

// Configuration for NatsJetStreamSource stage
message NatsJetStreamSourceConfig {
  // NATS server URL (default: nats://127.0.0.1:4222)
  string url = 1;

  // Subject to subscribe to (required)
  string subject = 2;

  // Stream name to bind to (optional)
  string stream_name = 3;

  // Durable consumer name (optional)
  string durable_name = 4;

  // Poll timeout in milliseconds (default: 1000)
  uint32 poll_timeout_ms = 5;
}
